### DISCUSSION TITLE
Completion not working (startCompletion is never answered)

### DISCUSSION THREAD

**Discussion by llucax** [neoclide/coc.nvim#2971] [Q&A]
Disclaimer: I have very little idea of how CoC works internally, and I only know very conceptually what a language server is.

Some time ago (couple of weeks?) I noticed completion wasn't always working. Sometimes it would just do nothing. Now it almost never work. All code hints, and popups while I write code showing function/class references work. But not code completion. Very seldom, usually the first time I ask for a completion pressing TAB after I opened nvim, it will work but then never again.

I checked the logs and see that when it works there is something like this in the logs:

* 2021-03-15T21:41:11.791 DEBUG (pid:194673) [completion] - trigger completion with { ... }
* 2021-03-15T21:41:12.117 INFO (pid:194673) [attach] - receive notification: startCompletion [ ... ]
* 2021-03-15T21:34:39.116 DEBUG (pid:190089) [completion-complete] - Source "around" takes 5ms
* 2021-03-15T21:34:39.116 DEBUG (pid:190089) [completion-complete] - Source "buffer" takes 3ms
* 2021-03-15T21:34:39.116 DEBUG (pid:190089) [completion-complete] - Source "file" takes 2ms
* 2021-03-15T21:34:39.318 DEBUG (pid:190089) [completion-complete] - Source "flutter-1" takes 211ms
* 2021-03-15T21:34:39.318 INFO (pid:190089) [completion-complete] - Results from: around,flutter-1
* 2021-03-15T21:34:39.323 DEBUG (pid:190089) [attach] - Notification autocmd: MenuPopupChanged { ... }
* etc.
* 2021-03-15T21:34:41.622 DEBUG (pid:190089) [attach] - Notification autocmd: CompleteDone { ... }

But when it doesn't, it's stuck in startCompletion:

* 2021-03-15T21:34:42.483 DEBUG (pid:190089) [completion] - trigger completion with { ... }
* 2021-03-15T21:34:42.827 INFO (pid:190089) [attach] - receive notification: startCompletion [ ... ]
* no more news

This happens with rust and dart/flutter. It seems to work as expected with json.

Any ideas of what could be going on or how to keep investigating?

Thanks!

**Comment by chemzqm** [comment]
https://github.com/neoclide/coc.nvim/wiki/Debug-coc.nvim

**Reply by llucax** [reply]
Thanks! I tried enabling the log to inspect communication between vim and coc.nvim but I can't make any sense from that's printed there either. I also tried using chrome devtools but again I don't even know what am I looking for.

Any pointers specifically on where I should look? What could be the most common reasons why this could happen?

**Comment by llucax** [comment]
Update: With some files it seems to work pretty well most of the time and with some files it seems to never work (or just one time as I said before).

**Reply by chemzqm** [reply]
The document won't be attched when it's too big or `&buftype` option is not normal buffer https://github.com/neoclide/coc.nvim/blob/master/src/completion/index.ts#L237

**Reply by llucax** [reply]
The files are very small, one is 139 LoC (with a maximum width of 100 chars). The buffers are just files opened regularly, the same as the other files that work (and that always worked).

I'm thinking of starting with an empty nvim config file and re-installing everything again and see if there is some configuration that is breaking stuff or what. If you have any other recommendations on how to narrow this down, it's more than welcome.
